### DISCUSSION TITLE
Is it fine that a system writes deep filtering warning message every time for author user?

### DISCUSSION THREAD

**Discussion by ByungJun25** [strapi/strapi#7317] [Legacy Topics]
Hello strapi teams,

Well, I just create a simple one element, it looks like below.

![image](https://user-images.githubusercontent.com/10557621/89311584-78d60b80-d676-11ea-88cc-e15b780f2bc6.png)

And I create one new user who has author as role.

![image](https://user-images.githubusercontent.com/10557621/89310046-7e325680-d674-11ea-80f6-328d5236640e.png)

And I give the authority to author user and I go to Testing1s collection types list view by author user.
(There is no data on Testing1 type)

![image](https://user-images.githubusercontent.com/10557621/89311751-afac2180-d676-11ea-9cb6-fd2628797b13.png)

**And now, when I see the log, I get the warning message about deep filtering although there is no data.**
(I guess, because of permission check?)

![image](https://user-images.githubusercontent.com/10557621/89311804-c2bef180-d676-11ea-951d-8b00eca9b21a.png)

I think it is not good for admin user because if there are several author user, then the admin only can see warning log.. 

Related code is here.
https://github.com/strapi/strapi/blob/35aafde2288e6858a9b1757cf2555a8561421140/packages/strapi-utils/lib/build-query.js#L142-L157

What do you think about it?? :)

**System**

- Node.js version: 12.16.3
- NPM version: 6.14.5
- Strapi version: 3.1.3
- Database: MongoDB
- Operating system: Windows 10 / Docker

Thank you
Byungjun Woo

**Comment by derrickmehaffy** [comment]
I think in this case it's not that big of an issue, that warning message is applied to every request that uses deep filtering (even if it's not a degraded performance query) The warning is only there until we review the database code in v4, for now you can safely ignore that warning as it really only applies to very complex and large deep filtering (which the admin won't do generally)

**Reply by ByungJun25** [reply]
Ah, okay Thank you @derrickmehaffy 
Then I will just ignore it. :)
